### PR TITLE
Buildfix: Visual Studio >= 14 supports timespec.

### PR DESCRIPTION
**Pull request by peters**

### CODE REVIEW

File: 3rdparty/include/pthreads/pthread.h [hunk 1]
```diff
@@ -317,6 +317,12 @@ enum {
 #endif
 #endif
 
+#if defined _MSC_VER && _MSC_VER >= 1900
+#ifndef HAVE_STRUCT_TIMESPEC
```
Comment (breznak) [inline 3rdparty/include/pthreads/pthread.h]: you can write these as one `#if`

File: 3rdparty/include/pthreads/pthread.h [hunk 2]
```diff
@@ -317,6 +317,12 @@ enum {
...
+#define HAVE_STRUCT_TIMESPEC 
+#endif
+#endif
+
 #if !defined(HAVE_STRUCT_TIMESPEC)
 #define HAVE_STRUCT_TIMESPEC
```
Comment (breznak) [inline 3rdparty/include/pthreads/pthread.h]: what is the purpose of this code? Is it correct?! 

```
if (??): define time_spec; 
if not time_spec: define time_spec; //!!!
```

**Review by breznak — COMMENTED** [review]
A suggestion for improvement & point out an (original) strange code

File: 3rdparty/include/pthreads/pthread.h (hunk 2 above)
Comment (breznak) [inline 3rdparty/include/pthreads/pthread.h]: Is this supposed to be the "header gate"? This low in the code..

**Comment by peters** [comment]
@nicehashdev Is there anything wrong with the PR since it has not been merged yet? :)

**Pull request closed without merging**